src/py21cmfast/src/SpinTemperatureBox.c

```diff
@@ -1949,29 +2084,60 @@ LOG_SUPER_DEBUG("looping over box...");
 
                             // next, Compton heating
                             //                dcomp_dzp = dT_comp(zp, T, x_e);
-                            dcomp_dzp = dcomp_dzp_prefactor*(x_e/(1.0+x_e+f_He))*( Trad_fast - T );
+                            dcomp_dzp = dcomp_dzp_prefactor*(x_e/(1.0+x_e))*( Trad_fast - T );//removed f_He
+
 
                             // lastly, X-ray heating
                             dxheat_dzp = dxheat_dt_box[box_ct] * dt_dzp * 2.0 / 3.0 / k_B / (1.0+x_e);
                             if (flag_options->USE_MINI_HALOS){
                                 dxheat_dzp_MINI = dxheat_dt_box_MINI[box_ct] * dt_dzp * 2.0 / 3.0 / k_B / (1.0+x_e);
                             }
 
+                            //CMB heating rate
+							dCMBheat_dzp = 0.;
+                            if (flag_options->USE_CMB_HEATING) {
+								//Meiksin et al. 2021
+								eps_CMB = (3./4.) * (T_cmb*(1.+zp)/T21) * A10_HYPERFINE * f_H * (hplank*hplank/Lambda_21/Lambda_21/m_p) * (1.+2.*T/T21);
```

Comment:
I can't find this equation. Can you be more specific please?

Reply:
The CMB heating term is actually computed using the last term in eq. 4 of  https://arxiv.org/pdf/2105.14516.pdf. You have to express P_CMB as (3/4)(T_CMB/T_*)A_10. Then you have to divide the whole term by (3/2) n_tot k_B, and have to apply cosmic-time to redshift conversion.